include/igl/fast_winding_number.cpp

```diff
@@ -4,321 +4,415 @@
+  Eigen::Matrix<int,Eigen::Dynamic,8> CH;
+  Eigen::Matrix<real,Eigen::Dynamic,3> CN;
+  Eigen::Matrix<real,Eigen::Dynamic,1> W;
 
+  octree(P,point_indices,CH,CN,W);
 
+  Eigen::Matrix<real,Eigen::Dynamic,Eigen::Dynamic> EC;
+  Eigen::Matrix<real,Eigen::Dynamic,3> CM;
+  Eigen::Matrix<real,Eigen::Dynamic,1> R;
 
+  fast_winding_number(P,N,A,point_indices,CH,expansion_order,CM,R,EC);
+  fast_winding_number(P,N,A,point_indices,CH,CM,R,EC,Q,beta,WN);
+}
 
+template <
+  typename DerivedP, 
+  typename DerivedA, 
+  typename DerivedN,
+  typename DerivedQ, 
+  typename DerivedWN>
+IGL_INLINE void igl::fast_winding_number(
+  const Eigen::MatrixBase<DerivedP>& P,
+  const Eigen::MatrixBase<DerivedN>& N,
+  const Eigen::MatrixBase<DerivedA>& A,
+  const Eigen::MatrixBase<DerivedQ>& Q,
+  Eigen::PlainObjectBase<DerivedWN>& WN)
+{
+  fast_winding_number(P,N,A,Q,2,2.0,WN);
+}
 
+template <
+  typename DerivedV,
+  typename DerivedF,
+  typename DerivedQ,
+  typename DerivedW>
+IGL_INLINE void igl::fast_winding_number(
+  const Eigen::MatrixBase<DerivedV> & V,
+  const Eigen::MatrixBase<DerivedF> & F,
+  const Eigen::MatrixBase<DerivedQ> & Q,
+  Eigen::PlainObjectBase<DerivedW> & W)
+{
+  igl::FastWindingNumberBVH fwn_bvh;
+  int order = 2;
+  igl::fast_winding_number(V,F,order,fwn_bvh);
+  float accuracy_scale = 2;
+  igl::fast_winding_number(fwn_bvh,accuracy_scale,Q,W);
+}
 
+template <
+  typename DerivedV,
+  typename DerivedF>
+IGL_INLINE void igl::fast_winding_number(
+  const Eigen::MatrixBase<DerivedV> & V,
+  const Eigen::MatrixBase<DerivedF> & F,
+  const int order,
+  FastWindingNumberBVH & fwn_bvh)
+{
+  assert(V.cols() == 3 && "V should be 3D");
+  assert(F.cols() == 3 && "F should contain triangles");
+  // Extra copies. Usuually this won't be the bottleneck.
+  fwn_bvh.U.resize(V.rows());
+  for(int i = 0;i<V.rows();i++)
+  {
+    for(int j = 0;j<3;j++)
+    {
+      fwn_bvh.U[i][j] = V(i,j);
+    }
+  }
+  // Wouldn't need to copy if F is **RowMajor**
+  fwn_bvh.F.resize(F.size());
+  for(int f = 0;f<F.rows();f++)
+  {
+    for(int c = 0;c<F.cols();c++)
+    {
+      fwn_bvh.F[c+f*F.cols()] = F(f,c);
+    }
+  }
+  fwn_bvh.ut_solid_angle.clear();
+  fwn_bvh.ut_solid_angle.init(
+     fwn_bvh.F.size()/3, 
+    &fwn_bvh.F[0], 
+     fwn_bvh.U.size(), 
+    &fwn_bvh.U[0], 
+    order);
+}
 
+template <
+  typename DerivedQ,
+  typename DerivedW>
+IGL_INLINE void igl::fast_winding_number(
+  const FastWindingNumberBVH & fwn_bvh,
+  const float _accuracy_scale,
+  const Eigen::MatrixBase<DerivedQ> & Q,
+  Eigen::PlainObjectBase<DerivedW> & W)
+{
+  const double accuracy_scale = 2;
+  assert(Q.cols() == 3 && "Q should be 3D");
+  W.resize(Q.rows(),1);
+  igl::parallel_for(Q.rows(),[&](int p)
+    {
+    FastWindingNumber::HDK_Sample::UT_Vector3T<float>Qp;
+          Qp[0] = Q(p,0);
+          Qp[1] = Q(p,1);
+          Qp[2] = Q(p,2);
+      W(p) = fwn_bvh.ut_solid_angle.computeSolidAngle(
+        Qp,
+        accuracy_scale) 
```

Comment:
_accuracy_scale not being used in computing solid angle. This blocks control of accuracy for this function and all that call it (triangle soup precomp+comp overload ln 334). 

Thought I'd point out simple (hopefully...) fix before merge.